src/index.ts

```diff
@@ -219,7 +219,7 @@ function createEmitterWithEvent<T, TEvent extends Event<T>>(
   func.event = event;
   func.emit = emit;
 
-  return func;
+  return Object.assign(func, [event, emit]);
```

Comment:
I don't think this will work, and even if it did I'm not a fan of using `Object.assign` because in some places like IE11 this library would then require a polyfill; and I'm trying to leave this dependency-less.

Object.assign won't assign prototype based properties, like `Array[Symbol.iterator]`. Because of this, during destructing it will result in runtime errors:
```
$ node
Welcome to Node.js v15.8.0.
Type ".help" for more information.
> function foo() {}
undefined
>
> foo
[Function: foo]
> Object.assign(foo, ["bar", "baz"])
[Function: foo] { '0': 'bar', '1': 'baz' }
> foo
[Function: foo] { '0': 'bar', '1': 'baz' }
> foo[0]
'bar'
> foo.length
0
> foo.concat
undefined
> const [a, ...b] = foo
Uncaught TypeError: foo is not iterable
```